### PR TITLE
Additional debugs to clarify error

### PR DESCRIPTION
**Pull request by bathina2**
## Change Overview

The 'Wrong type returned' error does not contain enough details. This logs some additional information.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [x] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/blockstorage/vmware/vmware.go [hunk 1]
```diff
@@ -124,6 +124,7 @@ func (p *FcdProvider) VolumeCreateFromSnapshot(ctx context.Context, snapshot blo
 	log.Debug().Print("CreateDiskFromSnapshot task complete", field.M{"VolumeID": volID, "SnapshotID": snapshotID})
 	obj, ok := res.(types.VStorageObject)
 	if !ok {
+		log.Error().Print("Wrong type returned for vSphere", field.M{"object": obj})
 		return nil, errors.New("Wrong type returned")
```
Comment (vkamra) [inline pkg/blockstorage/vmware/vmware.go:128]: Can we get this in the error? It'll get propagated up.

File: pkg/blockstorage/vmware/vmware.go [hunk 2]
```diff
@@ -124,6 +124,7 @@ func (p *FcdProvider) VolumeCreateFromSnapshot(ctx context.Context, snapshot blo
 	log.Debug().Print("CreateDiskFromSnapshot task complete", field.M{"VolumeID": volID, "SnapshotID": snapshotID})
 	obj, ok := res.(types.VStorageObject)
 	if !ok {
+		log.Error().Print("Wrong type returned for vSphere", field.M{"object": obj})
```
Comment (carlbraganza) [inline pkg/blockstorage/vmware/vmware.go:127]: May want to use Printf with %T?

Comment (carlbraganza) [reply pkg/blockstorage/vmware/vmware.go:127]: Should be dumping res and not obj. Obj would be nil

**Review by carlbraganza — APPROVED** [review]

File: pkg/blockstorage/vmware/vmware.go [hunk 3]
```diff
@@ -124,7 +125,7 @@ func (p *FcdProvider) VolumeCreateFromSnapshot(ctx context.Context, snapshot blo
 	log.Debug().Print("CreateDiskFromSnapshot task complete", field.M{"VolumeID": volID, "SnapshotID": snapshotID})
 	obj, ok := res.(types.VStorageObject)
 	if !ok {
-		return nil, errors.New("Wrong type returned")
+		return nil, errors.New(fmt.Sprintf("Wrong type returned for vSphere. Type: %T, Value: %v", res, res))
```
Comment (julio-lopez) [inline pkg/blockstorage/vmware/vmware.go:128]: @bathina2 Would `errors.Errorf` work here?

https://pkg.go.dev/github.com/pkg/errors#Errorf

**Pull request merged**